File: 백준/Silver/13305. 주유소/주유소.py
n = int(input())
road = list(map(int,input().split()))
oil = list(map(int,input().split()))
min_oil = oil[0]
ans = oil[0] * road[0]

for i in range(1,len(road)):
    if oil[i] < min_oil:
        min_oil = oil[i]
        ans += road[i] * min_oil
    else:
        ans += road[i] * min_oil
print(ans)

File: 백준/Silver/13305. 주유소/주유소.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n  = Integer.parseInt(br.readLine());
        String[] tempToll = br.readLine().split(" ");
        String[] tempOil = br.readLine().split(" ");
        int[] km = new int[n-1];
        int[] oil = new int[n];

        for (int i=0; i<n; i++){
            if (i<n-1){
                km[i] = Integer.parseInt(tempToll[i]);
            }
            oil[i] = Integer.parseInt(tempOil[i]);
        }

        int minOil = oil[0] ;
        int ans = minOil * km[0];

        for (int i=1; i<n-1; i++){
            if (minOil >oil[i]){
                minOil = oil[i];
                ans += (minOil* km[i]);
            }else {
                ans += (minOil * km[i]);
            }
        }
        System.out.println(ans);
    }
}